### ISSUE TITLE
[Request] Add, remove, reorder messages and assign/reassign roles

### ISSUE THREAD

**Issue by Algorithm5838** [lobehub/lobehub#2065]
### 🥰 Feature Description

Currently, in lobe-chat, there is no way to modify the conversation history or reassign roles (user, assistant, system) to existing messages.

### 🧐 Proposed Solution

Implement methods to:
1. Add new messages to the conversation history
2. Remove existing messages from the conversation history
3. Reorder messages in the conversation history
4. Assign or reassign roles (user, assistant, system) to existing messages

### 📝 Additional Information

This feature would greatly improve the usability and flexibility of lobe-chat, allowing users to curate and modify conversation histories as needed. Similar functionality is already implemented in other projects like BetterChatGPT and NextChat.

**Comment by arvinxx** [comment]
> Remove existing messages from the conversation history

there is a delete button

**Comment by Algorithm5838** [comment]
You're right, I was considering a separate settings UI that would handle the features I mentioned. However, you could implement them within the same UI as well.
